google/cloud/storage/internal/insert_object_media_request.h

```diff
@@ -29,10 +29,11 @@ namespace internal {
  * TODO(#710) - add missing request parameters.
  */
 class InsertObjectMediaRequest
-    : public GenericRequest<InsertObjectMediaRequest, Generation,
-                            IfGenerationMatch, IfGenerationNotMatch,
+    : public GenericRequest<InsertObjectMediaRequest, ContentEncoding,
+                            Generation, IfGenerationMatch, IfGenerationNotMatch,
                             IfMetaGenerationMatch, IfMetaGenerationNotMatch,
-                            Projection, UserProject> {
+                            KmsKeyName, PredefinedAcl, Projection, UploadType,
```

Comment:
I do not think `UploadType` should be modifiable by the application. We need it internally.  Please remove from this list (you can leave the class if you want).